src/Generator/Synapse/ImportIncremental/FromAbsGenerator.php

```diff
@@ -0,0 +1,191 @@
+
+use Doctrine\DBAL\Connection;
+use Keboola\CsvOptions\CsvOptions;
+use Keboola\CustomQueryManagerApp\Generator\GeneratorInterface;
+use Keboola\CustomQueryManagerApp\Generator\Replace;
+use Keboola\CustomQueryManagerApp\Generator\ReplaceToken;
+use Keboola\CustomQueryManagerApp\Generator\Utils;
+use Keboola\Datatype\Definition\BaseType;
+use Keboola\Datatype\Definition\Synapse;
+use Keboola\Db\ImportExport\Backend\Synapse\SynapseImportOptions;
+use Keboola\Db\ImportExport\Backend\Synapse\ToFinalTable\IncrementalImporter;
+use Keboola\Db\ImportExport\Backend\Synapse\ToStage\ToStageImporter;
+use Keboola\Db\ImportExport\Storage;
+use Keboola\TableBackendUtils\Column\ColumnCollection;
+use Keboola\TableBackendUtils\Column\ColumnInterface;
+use Keboola\TableBackendUtils\Column\SynapseColumn;
+use Keboola\TableBackendUtils\Escaping\SynapseQuote;
+use Keboola\TableBackendUtils\Table\Synapse\TableDistributionDefinition;
+use Keboola\TableBackendUtils\Table\Synapse\TableIndexDefinition;
+use Keboola\TableBackendUtils\Table\SynapseTableDefinition;
+use Keboola\TableBackendUtils\Table\SynapseTableQueryBuilder;
+use PHPUnit\Framework\TestCase;
+
+class FromAbsGenerator extends TestCase implements GeneratorInterface
+{
+    /**
+     * @param string[] $columns
+     * @param string[] $primaryKeys
+     * @return string[]
+     */
+    public function generate(array $columns, array $primaryKeys = []): array
+    {
+        $sourceColumns = $columns;
+
+        $stageColumns = [];
+        foreach ($columns as $columnName) {
+            $stageColumns[] = new SynapseColumn(
+                $columnName,
+                new Synapse(Synapse::getTypeByBasetype(BaseType::STRING))
+            );
+        }
+        $stagePrimaryKeys = $primaryKeys;
+
+        $destColumns = $stageColumns;
+        $destColumns[] = new SynapseColumn(
+            ColumnInterface::TIMESTAMP_COLUMN_NAME,
+            new Synapse(Synapse::getTypeByBasetype(BaseType::TIMESTAMP))
+        );
+        $destPrimaryKeys = $primaryKeys;
+
+        $params = [
+            'sourceFiles' => [
+                new ReplaceToken(
+                    Utils::getUniqeId('sourceFile1'),
+                    'listFiles(sourceFiles)',
+                    Replace::TYPE_MATCH_AS_VALUE_CUSTOM,
+                ),
+            ],
+            'sourceContainerUrl' => new ReplaceToken(
+                Utils::getUniqeId('sourceContainerUrl'),
+                'sourceContainerUrl',
+                Replace::TYPE_MATCH_AS_VALUE,
+            ),
+
+            'stageTableName' => new ReplaceToken(
+                Utils::getUniqeId('__temp_stageTableName'),
+                'stageTableName',
+            ),
+            // dedup table (prefix)
+            'dedup_stageTableName' => new ReplaceToken(
+                '#__temp_csvimport',
+                "destTableName ~ rand ~ '_tmp'",
+                Replace::TYPE_PREFIX_AS_IDENTIFIER,
+            ),
```

Comment:
nemel by byt tady a na par dalsich mistech taky `#` pro TEMP tabulku?
Bud `'#' ~ destTableName ~ rand ~ '_tmp'` nebo `stageTableName ~ '_dedup'`

Reply:
To sa liší usecase od usecase, pokud je tam ten rename u full loadu tak tam musíja byt permanentní tabulky. Když je to dedup tabulka tak sa používá temp tabulka.
když sa podíváš na full load tak tam sa nepužívá temp tabulka pro dedup ani nikde jinde krom stage https://github.com/keboola/php-db-import-export/blob/master/src/Backend/Synapse/ToFinalTable/FullImporter.php
Když sa podíváš na incremental tak tam sa používá dedup tabulka tmp, protože tam není ten swap pak. https://github.com/keboola/php-db-import-export/blob/master/src/Backend/Synapse/ToFinalTable/IncrementalImporter.php#L55

Reply:
Jo, prave tu incremental dedup tabulku jsem mel na mysli, tam by to mohla byt ta `stageTableName ~ '_dedup'`.
Nevim presne, jak se lisi temp tabulka od normalni...

Reply:
Synapse má nějaků lokální a remote storage, temp tabulka je lokální takže je rychlejší a existuje enom pro session. Klidně ty dedup tabulky pro incremental tak udělaj.